DigitalLearningSolutions.Web/Scripts/trackingSystem/reports.ts

```diff
@@ -71,24 +75,37 @@ request.open('GET', path, true);
 request.responseType = 'json';
 request.send();
 
-let allRowsShown = false;
-const viewMoreLink = <HTMLLIElement>document.getElementsByClassName('js-shown-load-more').item(0);
-viewMoreLink.style.display = 'block';
-
 viewMoreLink.addEventListener('click', (event) => {
   event.preventDefault();
-  allRowsShown = !allRowsShown;
-  showMoreRows(allRowsShown);
-  viewMoreLink.innerText = allRowsShown ? 'View Less' : 'View More';
+  const activityRow = <HTMLElement>document.getElementsByClassName(activityToggleRowClass).item(0);
+
+  if (activityRow?.style.display === activityToggleableRowDisplayNone) {
+    viewMoreRows();
+    viewMoreLink.innerText = 'View Less';
+  } else {
+    viewLessRows();
+    viewMoreLink.innerText = 'View More';
+  }

```

Comment:
It'd be nice to extract this into a function `toggleVisibleActivityRows()` or such like.